File: Python/31-40/e_34.py
# Crea en python un bucle que contenga los números del 10 al 20, dentro de ese bucle crea otro que repita la misma lista 10 veces. Imprime el resultado.


# Bucle principal con los números del 10 al 20
for numero_principal in range(10, 21):
    # Lista del 10 al 20
    lista_numeros = list(range(10, 21))

    # Bucle interno que repite la lista 10 veces
    for _ in range(10):
        # Imprime la lista
        print(lista_numeros)

    # Imprime el número principal
    print(f"Número principal: {numero_principal}")
    print("=" * 20)  # Línea separadora


# Este código utiliza dos bucles, donde el bucle principal itera a través de los números del 10 al 20. Dentro de este bucle,
# hay otro bucle interno que repite la misma lista del 10 al 20 diez veces.
# La estructura de impresión proporciona una salida detallada con el número principal y la lista repetida, pero todo se imprime solo una vez.

File: JS/31-40/e_34.js
// Bucle principal con los números del 10 al 20
for (let numeroPrincipal = 10; numeroPrincipal <= 20; numeroPrincipal++) {
    // Lista del 10 al 20
    let listaNumeros = Array.from({ length: 11 }, (_, index) => index + 10);

    // Bucle interno que repite la lista 10 veces
    for (let i = 0; i < 10; i++) {
        // Imprime la lista
        console.log(listaNumeros);
    }

    // Imprime el número principal
    console.log(`Número principal: ${numeroPrincipal}`);
    console.log("=".repeat(20));  // Línea separadora
}


// Este código utiliza un bucle for para iterar a través de los números del 10 al 20.
// Dentro de este bucle, hay otro bucle interno que crea la lista del 10 al 20 y la imprime 10 veces.
// La impresión del número principal y la línea separadora se realiza una vez fuera del bucle interno.